pkg/monitortests/authentication/nodefaultserviceaccountoperatorstests/monitortest.go

```diff
@@ -0,0 +1,203 @@
+		return strings.HasPrefix(pod.Name, "kube-apiserver-guard-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "kube-controller-manager-guard-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "openshift-kube-scheduler-guard-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "monitoring-plugin-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "multus-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "networking-console-plugin-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "network-check-target-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "verify-all-openshiftcommunityoperators-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "verify-all-openshiftredhatmarketplace-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "verify-all-openshiftcertifiedoperators-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "verify-all-openshiftredhatoperators-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "version-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "must-gather-")
+	},
+	func(pod corev1.Pod) bool {
+		return pod.Namespace == "openshift-marketplace"
+	},
+}
+
+// generateDefaultSAFailures generates a list of failures where the pod in a list of pods
+// violated the default service account check.
+func generateDefaultSAFailures(podList []corev1.Pod) []*junitapi.JUnitTestCase {
+	junits := []*junitapi.JUnitTestCase{}
+	failures := make([]string, 0)
+	for _, pod := range podList {
+		podSA := pod.Spec.ServiceAccountName
+		// if the service account name is not default, we can exit for that iteration
+		if podSA != "default" {
+			continue
+		}
+		hasException := false
+		for _, exception := range exceptions {
+			if exception(pod) {
+				hasException = true
+				break
+			}
+		}
+		if hasException {
+			// flag exception as flaky failure
+			failures = append(failures, fmt.Sprintf("[flake] service account name %s is being used in pod %s in namespace %s", podSA, pod.Name, pod.Namespace))
+			continue
+		}
+		// otherwise, we need to flag the failure
+		failures = append(failures, fmt.Sprintf("service account name %s is being used in pod %s in namespace %s", podSA, pod.Name, pod.Namespace))
+		// generate tests for given namespace/pod
+		testName := fmt.Sprintf("[sig-auth] pod '%s/%s' must not use the default service account", pod.Namespace, pod.Name)
+		if len(failures) == 0 {
+			junits = append(junits, &junitapi.JUnitTestCase{Name: testName})
+			continue
+		}
+		failureMsg := strings.Join(failures, "\n")
+		junits = append(junits, &junitapi.JUnitTestCase{
+			Name:          testName,
+			SystemOut:     failureMsg,
+			FailureOutput: &junitapi.FailureOutput{Output: failureMsg},
+		})
+		if strings.Contains(failureMsg, "[flake]") {
```

Comment:
I'd move this to being appended to when we find out that this is an exemption case instead of doing string parsing/matching